src/web/core/src/main/java/org/geoserver/web/data/resource/InternationalStringPanel.java

```diff
@@ -257,12 +272,25 @@ private GrowableInternationalString updateGrowableString() {
         GrowableInternationalString growableString = new GrowableInternationalString();
         List<GrowableStringModel.InternationalStringEntry> items = provider.getItems();
         for (GrowableStringModel.InternationalStringEntry entry : items) {
-            growableString.add(entry.getLocale(), entry.getText());
+            if (canAddElement(items, entry)) growableString.add(entry.getLocale(), entry.getText());
         }
         growableModel.setObject(growableString);
         return growableString;
     }
 
+    private boolean canAddElement(
```

Comment:
Can you explain what is happening here?

Reply:
The support of null locale, forced the validation for double null locale entries to be delayed after the wicket validation phase, doing it before the submit (during wicket validation phase in the IValidator, it is not possible to tell when a Locale is null because a new entry was added or because it is meant to be null).
This in turn causes the fact that when the user plays with the UI a newly added field with null locale entry might be added to the GrowableString. Before the null locale support the null values where filter out before adding entires to the Growable instance, but now this might cause entries with null key to be duplicated and make the Growable throw an exception.
I've added a clarifying comment over the method.